src/librustc_typeck/collect/type_of.rs

```diff
@@ -188,12 +188,12 @@ pub(super) fn type_of(tcx: TyCtxt<'_>, def_id: DefId) -> Ty<'_> {
         Node::Field(field) => icx.to_ty(&field.ty),
 
         Node::Expr(&Expr { kind: ExprKind::Closure(.., gen), .. }) => {
-            if gen.is_some() {
-                return tcx.typeck_tables_of(def_id).node_type(hir_id);
-            }
```

Comment:
That's funny, I almost did the opposite last week, trying to make #69968 work.
(It was a bad idea and the solution was to fix the actual code that couldn't handle unsubstitued closure types)